sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -470,53 +476,75 @@ private static boolean isPortResource(Protos.Resource resource) {
 
         // command and user:
 
-        Protos.CommandInfo.Builder commandInfoBuilder = executorInfoBuilder.getCommandBuilder()
-                .setValue("export LD_LIBRARY_PATH=$MESOS_SANDBOX/libmesos-bundle/lib:$LD_LIBRARY_PATH && " +
-                        "export MESOS_NATIVE_JAVA_LIBRARY=$(ls $MESOS_SANDBOX/libmesos-bundle/lib/libmesos-*.so) && " +
-                        "export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/) && " +
-                        "./executor/bin/executor");
+        Protos.CommandInfo.Builder executorCommandBuilder = executorInfoBuilder.getCommandBuilder().setValue(
+                "export LD_LIBRARY_PATH=$MESOS_SANDBOX/libmesos-bundle/lib:$LD_LIBRARY_PATH && " +
+                "export MESOS_NATIVE_JAVA_LIBRARY=$(ls $MESOS_SANDBOX/libmesos-bundle/lib/libmesos-*.so) && " +
+                "export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/) && " +
+                "./executor/bin/executor");
 
         if (podSpec.getUser().isPresent()) {
-            commandInfoBuilder.setUser(podSpec.getUser().get());
+            executorCommandBuilder.setUser(podSpec.getUser().get());
         }
 
         // URIs:
 
+        // Required in scheduler env.
         String executorUri = System.getenv(EXECUTOR_URI);
         if (executorUri == null) {
-            throw new IllegalStateException("Missing environment variable: " + EXECUTOR_URI);
+            throw new IllegalStateException("Missing required environment variable: " + EXECUTOR_URI);
         }
-        commandInfoBuilder.addUrisBuilder().setValue(executorUri);
+        executorCommandBuilder.addUrisBuilder().setValue(executorUri);
 
+        // Required in scheduler env.
         String libmesosUri = System.getenv(LIBMESOS_URI);
         if (libmesosUri == null) {
-            throw new IllegalStateException("Missing environment variable: " + LIBMESOS_URI);
+            throw new IllegalStateException("Missing required environment variable: " + LIBMESOS_URI);
         }
-        commandInfoBuilder.addUrisBuilder().setValue(libmesosUri);
+        executorCommandBuilder.addUrisBuilder().setValue(libmesosUri);
 
-        // Reuse scheduler's JAVA_URI for executors when available:
+        // Reuse scheduler's JAVA_URI for executors when available, or fall back to default.
         String javaUri = System.getenv(JAVA_URI);
         if (javaUri == null) {
             javaUri = DEFAULT_JAVA_URI;
         }
-        commandInfoBuilder.addUrisBuilder().setValue(javaUri);
+        executorCommandBuilder.addUrisBuilder().setValue(javaUri);
 
-        for (TaskSpec taskSpec : podSpec.getTasks()) {
-            if (!taskSpec.getCommand().isPresent()) {
-                continue;
+        // Any URIs defined in PodSpec itself.
+        if (podSpec.getUris() != null) {
```

Comment:
Why do we have null checks here and below?  Shouldn't these be guaranteed to be empty collections?